task/control-plane-agent/src/mgs_common.rs

```diff
@@ -143,6 +147,108 @@ impl MgsCommon {
             CabooseError::NoImageHeader => panic!(),
         })
     }
+
+    pub(crate) fn reset_component_prepare(
+        &mut self,
+        component: SpComponent,
+    ) -> Result<(), SpError> {
+        // TODO: Add some kind of auth check before performing a reset.
+        // https://github.com/oxidecomputer/hubris/issues/723
+        ringbuf_entry!(Log::MgsMessage(MgsMessage::ResetPrepare));
+        self.reset_component_requested = component;
+        Ok(())
+    }
+
+    pub(crate) fn reset_component_trigger(
+        &mut self,
+        update: &SpUpdate,
+        component: SpComponent,
+        slot: Option<u16>,
```

Comment:
It looks like this API numbers slots in a component-specific manner starting from 0 using some subset of the `u16` space -- this appears to be the only place where this is done, could you discuss it in the rustdoc comment?

It would also be worth mentioning that the argument is completely ignored if the target is anything other than the RoT. (Arguments that are completely ignored if certain other arguments have certain other values should almost always be fields on an enum, fwiw.)